modules/openapi-generator/src/main/resources/typescript-axios/configuration.mustache

```diff
@@ -78,13 +78,17 @@ export class Configuration {
         this.accessToken = param.accessToken;
         this.basePath = param.basePath;
         this.serverIndex = param.serverIndex;
+        {{#httpUserAgent}}
         this.baseOptions = {
+            ...param.baseOptions,
             headers: {
                 ...param.baseOptions?.headers,
-                'User-Agent': "OpenAPI-Generator{{#npmVersion}}/{{npmVersion}}{{/npmVersion}}/typescript-axios"
+                'User-Agent': "{{httpUserAgent}}",
```

Comment:
lets simplify the template and just wrap this line here with `{{#httpUserAgent}}`
```suggestion
                {{#httpUserAgent}}
                'User-Agent': "{{httpUserAgent}}",
                {{/httpUserAgent}}
```

and also, can you swap it with `...param.baseOptions?.headers,`, so it can always be overwritten by the config headers?

Reply:
Oh yeah, good call.